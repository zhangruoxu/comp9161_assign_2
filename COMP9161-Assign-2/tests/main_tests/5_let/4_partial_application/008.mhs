main 
     =
 let x = 8;

 -- do they correctly get the environment for the partial eval?
 in let 
     f =
        letfun f w =
          letfun g y =
               y + x + w + 1;

 in f 3 2;
