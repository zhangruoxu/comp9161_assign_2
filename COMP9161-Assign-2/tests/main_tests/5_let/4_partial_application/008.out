main :: Int = let x :: Int = 8;
              in let f :: (Int -> Int -> Int) = (letfun f :: (Int -> Int -> Int) w = (letfun g :: (Int -> Int) y = (+) ((+) ((+) y x) w) 1));
                 in f 3 2;
